### PR TITLE
#182282568 : INITIALIZE-PROJECT

### PR DESCRIPTION
**Pull request by foscraft**
#### What does this Pull Request do?
+ initialize project
#### Description of Task to be completed?
+ start django project
+ configure environment
+ configure postgresql database 
#### How should this be manually tested?
+ unit tests for database connection using pytest
#### Any background context you want to provide?
N/A
#### What are the relevant Pivotal tracker card?
+ Start django project
#### Screenshots (if appropriate)
+  rc
#### Questions:
N/A

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+    ignore = E203, E266, E501, W503, F403, F401
```
Comment (mainamuriuki274) [inline .flake8]: Don't ignore these ones F403, F401, apart from the settings init file

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -4,4 +4,4 @@
 #### Any background context you want to provide?
 #### What are the relevant Pivotal tracker card?
 #### Screenshots (if appropriate)
-#### Questions:
\ No newline at end of file
+### Questions:
```
Comment (mainamuriuki274) [inline .github/pull_request_template.md]: Why have you removed one #

File: core/settings/local.py [hunk 3]
```diff
@@ -0,0 +1,56 @@
+"""
```
Comment (mainamuriuki274) [inline core/settings/local.py]: Remove unneccessary comment

File: core/settings/local.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+DEBUG = True
+
+ALLOWED_HOSTS = [
+    "localhost",
+]
+
+# Application definition
+
+
+MIDDLEWARE = [
```
Comment (mainamuriuki274) [inline core/settings/local.py]: Already defined in base.py

File: core/settings/local.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+
+# Database
+# https://docs.djangoproject.com/en/4.0/ref/settings/#databases
+
+
+# Password validation
+# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators
+
+AUTH_PASSWORD_VALIDATORS = [
```
Comment (mainamuriuki274) [inline core/settings/local.py]: Already defined in base.py

File: core/settings/local.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/4.0/ref/settings/
+"""
+
+import os
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = True
+
+ALLOWED_HOSTS = [
```
Comment (mainamuriuki274) [inline core/settings/local.py]: Supposed to add from the ones in base.py i.e ALLOWED_HOSTS += ["localhost"]

File: core/settings/production.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+For the full list of settings and their values, see
+https://docs.djangoproject.com/en/4.0/ref/settings/
+"""
+
+import os
+
+# SECURITY WARNING: don't run with debug turned on in production!
+DEBUG = False
+
+ALLOWED_HOSTS = [""]
```
Comment (mainamuriuki274) [inline core/settings/production.py]: Unneccessary

File: core/settings/production.py [hunk 8]
```diff
@@ -0,0 +1,54 @@
+"""
```
Comment (mainamuriuki274) [inline core/settings/production.py]: Remove unnecessary comments

File: core/settings/production.py [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+
+# Database
+# https://docs.djangoproject.com/en/4.0/ref/settings/#databases
+
+
+# Password validation
+# https://docs.djangoproject.com/en/4.0/ref/settings/#auth-password-validators
+
+AUTH_PASSWORD_VALIDATORS = [
```
Comment (mainamuriuki274) [inline core/settings/production.py]: Allready in base.py

File: core/settings/staging.py [hunk 10]
```diff
@@ -0,0 +1,54 @@
+"""
```
Comment (mainamuriuki274) [inline core/settings/staging.py]: Remove unneccessary code and comments, dont repeat things that are in base.py if you are not changing them

File: mypy.ini [hunk 11]
```diff
@@ -0,0 +1,12 @@
+[mypy]
+plugins = mypy_django_plugin.main, mypy_drf_plugin.main
+ignore_missing_imports = True
+warn_unused_ignores = True
+warn_redundant_casts = True
+strict_optional = False
+
+[mypy.plugins.django-stubs]
+django_settings_module = "core.settings.local"
```
Comment (mainamuriuki274) [inline mypy.ini]: Just point it to settings withoout .local

File: requirements.txt [hunk 12]
```diff
@@ -0,0 +1,162 @@
+altair==4.2.0
```
Comment (mainamuriuki274) [inline requirements.txt]: This requirements file has packages we do not need that is why we use the virtual environemnt

**Review by mainamuriuki274 — CHANGES_REQUESTED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (foscraft) [reply .github/pull_request_template.md]: I will be more careful next time, I deleted by mistake.

**Review by foscraft — COMMENTED** [review]
All comments resolved

File: .github/pull_request_template.md (hunk 2 above)
Comment (foscraft) [reply .github/pull_request_template.md]: All comments resolved

File: core/settings/local.py (hunk 3 above)
Comment (foscraft) [reply core/settings/local.py]: Done

File: core/settings/staging.py (hunk 10 above)
Comment (foscraft) [reply core/settings/staging.py]: Done

**Review by foscraft — COMMENTED** [review]
All comments resolved

**Review by foscraft — COMMENTED** [review]
Comments resolved

**Review by foscraft — COMMENTED** [review]
Resolved the requested changes

**Review by foscraft — COMMENTED** [review]
Resolved the requested changes

**Pull request merged**